File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.js
function solution(video_len, pos, op_start, op_end, commands) {
  const videoLenS = toSecond(video_len);
  let posS = toSecond(pos);
  const opStartS = toSecond(op_start);
  const opEndS = toSecond(op_end);

  if (posS >= opStartS && posS <= opEndS) posS = opEndS;

  for (let i = 0; i < commands.length; i++) {
    posS += commands[i] === "next" ? 10 : -10;
    if (posS < 0) {
      posS = 0;
    }
    if (posS > videoLenS) {
      posS = videoLenS;
    }
    if (posS >= opStartS && posS <= opEndS) {
      posS = opEndS;
    }
  }

  const h = Math.floor(posS / 60) + "";
  const m = (posS % 60) + "";

  return `${h.padStart(2, "0")}:${m.padStart(2, "0")}`;
}

function toSecond(target) {
  const [minute, second] = target.split(":");
  return Number(minute) * 60 + Number(second);
}

console.log(solution("34:33", "13:00", "00:55", "02:55", ["next", "prev"]));

File: 프로그래머스/1/340213. ［PCCP 기출문제］ 1번 ／ 동영상 재생기/［PCCP 기출문제］ 1번 ／ 동영상 재생기.py
from typing import List

def to_second(value: str) -> int:
    minute, second = map(int, value.split(':'))
    return minute * 60 + second

def to_time(second: int) -> str:
    minute, sec = divmod(second, 60)
    return f"{minute:02d}:{sec:02d}"

def solution(video_len: str, pos: str, op_start: str, op_end: str, commands: List[str]) -> str:
    video_length = to_second(video_len)
    current = to_second(pos)
    op_start_sec = to_second(op_start)
    op_end_sec = to_second(op_end)

    if op_start_sec <= current <= op_end_sec:
        current = op_end_sec

    for command in commands:
        if command == 'next':
            current += 10
        else:
            current -= 10
        current = min(max(current, 0), video_length)
        if op_start_sec <= current <= op_end_sec:
            current = op_end_sec
    return to_time(current)
